src/wave.js

```diff
@@ -336,4 +338,29 @@ export class Wave extends mix(WaveBase).with(
     doFunc(func) {
         func(this);
     } // for scripting
+
+    async takeGifScreenshot(options = {}) {
+        try {
+            const gifDataUrl = await createRotatingGif(this, options);
+
+            // Use custom filename from options or fall back to default
+            const fileName = options.downloadPath
+                ? options.downloadPath.split("/").pop() // Extract filename from path
+                : (this._structure.name || this._structure.formula || "wave-visualization") +
+                  ".gif";
+
+            // Download the GIF
+            const a = document.createElement("a");
+            a.href = gifDataUrl;
+            a.download = fileName;
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
```

Comment:
I believe we already have this function for downloading data URL in utils or in cove